File: client/src/app/modules/miscellaneous/services/weekday.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { WeekDay } from '../../miscellaneous/model/WeekDay';

@Injectable({
  providedIn: 'root'
})
export class WeekDayService extends AppRestService<WeekDay>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getWeekDays() {
    return this.getListCached(new HttpParams(), "weekdays");
  }

  addOrUpdateWeekDay(weekDay: WeekDay) {
    this.clearListCache(new HttpParams(), "weekdays");
    return this.addOrUpdate(new HttpParams(), "weekday", weekDay, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/WeekDayService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.WeekDay;

public interface WeekDayService {
    public List<WeekDay> getWeekDays();

    public WeekDay getWeekDay(Integer id);

    public WeekDay addOrUpdateWeekDay(WeekDay weekDay);
}
